### PR TITLE
Adds plasmamen as a playable race 2.0

### PR DESCRIPTION
**Pull request by as334**
See [This post](https://tgstation13.org/phpBB/viewtopic.php?f=10&t=4828) for details. Please direct non-code related feedback over there as well, since I was commissioned to code this. What's currently left to do is mostly just testing stuff, and fixing any bugs that show up. 

:cl:
rscadd: Nanotrasen has hired a brand new supply of ~~expendable labor~~ _LOYAL CREW MEMBERS_ please welcome them with open arms.
rscadd: Plasmamen are now a playable race. They require plasma gas to survive and will ignite in oxygen environments.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/species_types.dm [hunk 1]
```diff
@@ -450,6 +453,28 @@ var/global/image/plasmaman_on_fire = image("icon"='icons/mob/OnFire.dmi', "icon_
...
+	return 0
 
+/datum/species/plasmaman/qualifies_for_rank(rank, list/features)
+	if(rank in command_positions)
+		return 0
+	if(rank in security_positions)
+		return 0
+	if(rank == "Clown" || rank == "Mime")//No funny bussiness
+		return 0
+	if(rank == "Bartender")
```
Comment (Anonus) [inline code/modules/mob/living/carbon/human/species_types.dm]: >no flaming barmen

why

Comment (as334) [inline code/modules/mob/living/carbon/human/species_types.dm]: >Please direct feedback to the feedback thread

File: code/datums/supplypacks.dm [hunk 2]
```diff
@@ -1177,6 +1177,16 @@ var/list/all_supply_groups = list(supply_emergency,supply_security,supply_engine
 	cost = 20
 	containername = "high-traction floor tiles"
 
+/datum/supply_packs/misc/plasmaman
+	name = "Plasma-man Supply Kit"
+	contains = list(/obj/item/clothing/under/plasmaman,
+					/obj/item/clothing/under/plasmaman,
+					/obj/item/weapon/tank/internals/plasmaman/belt/full,
+					/obj/item/weapon/tank/internals/plasmaman/belt/full,
```
Comment (Anonus) [inline code/datums/supplypacks.dm]: Doesn't exist, thus the checks failing

File: code/datums/supplypacks.dm [hunk 3]
```diff
@@ -1177,6 +1177,16 @@ var/list/all_supply_groups = list(supply_emergency,supply_security,supply_engine
...
 	containername = "high-traction floor tiles"
 
+/datum/supply_packs/misc/plasmaman
+	name = "Plasma-man Supply Kit"
+	contains = list(/obj/item/clothing/under/plasmaman,
+					/obj/item/clothing/under/plasmaman,
+					/obj/item/weapon/tank/internals/plasmaman/belt/full,
+					/obj/item/weapon/tank/internals/plasmaman/belt/full,
+					/obj/item/clothing/head/helmet/space/hardsuit/plasmaman,
+					/obj/item/clothing/head/helmet/space/hardsuit/plasmaman)
```
Comment (Anonus) [inline code/datums/supplypacks.dm]: >10 point suits

A single suit is 80 points, and while that might be too much imo this is also kind of eh.

File: code/modules/mob/living/carbon/human/species_types.dm (hunk 1 above)
Comment (SweaterKittens) [inline code/modules/mob/living/carbon/human/species_types.dm]: @Anonus: The thought was to put them in purely work-oriented positions for starters, as players/admins were concerned about the griefing that might ensue. However if there's enough interest to have them as Bartenders I don't really see that being a big problem, although I'd prefer to keep them out of Clown and Mime.

File: code/modules/clothing/under/miscellaneous.dm [hunk 4]
```diff
@@ -489,3 +489,54 @@
...
+			if(next_extinguish > world.time)
+				return
+			next_extinguish = world.time + extinguish_cooldown
+			extinguishes_left--
+			H.visible_message("<span class='warning'>[H]'s suit automatically extinguishes them!</span>","<span class='warning'>Your suit automatically extinguishes you.</span>")
+			H.ExtinguishMob()
+			PoolOrNew(/obj/effect/effect/water, get_turf(H))
+	return 0
+
+/obj/item/clothing/under/plasmaman/attackby(obj/item/device/extinguisher_refill/E, mob/user, params)
```
Comment (optimumtact) [inline code/modules/clothing/under/miscellaneous.dm]: is it even possible to specify the full type path in the method like this and have it work instead of just casting every atom you get attackedby?

Comment (RemieRichards) [inline code/modules/clothing/under/miscellaneous.dm]: Yeah this will refill the suit with EVERY SINGLE ITEM YOU HIT IT WITH.
not just extinguishers, and not just extinguishers with contents either.

Comment (as334) [inline code/modules/clothing/under/miscellaneous.dm]: I'll  be honest this was the last bit I coded and I just tossed it together last minute. I'll make it work properly in the first round of fixes

**Comment by Dorsisdwarf** [comment]
as334, is this going to be fixed any time soon?

It doesnt even compile

**Comment by as334** [comment]
I fixed most of the issues, i've just got to do a bit more bug fixing, and then i'll commit it. But yes it is actively being worked on.

**Comment by as334** [comment]
Kill me now.

**Comment by as334** [comment]
I'm gonna wait till i actually get some sprites in order to fix the merge conflicts, otherwise it'll lead to a lot of commit spam.

**Comment by chimersable** [comment]
I think if you have to restrict usage because of grief potential then they are probably over powered. Tone it down.

**Comment by Razharas** [comment]
@as334

**Comment by SweaterKittens** [comment]
@chimersable Like As334, any non-code related feedback should be directed towards my forum thread. But I will say, if you look at what they bring to the table I'm sure you will agree that they are not in any way overpowered. Clown/Mime are just very griefy roles in general, and I would prefer to not have Plasmamen running around being shitters when the community is already cautious about nonhuman races as is, after lizards. Perhaps they can be enabled at a later date, but not for the initial implementation.

**Comment by WJohn** [comment]
So about forever later I finish the plasmamen jumpsuits. I got swamped with life stuff, alongside being very generally lazy. It's no excuse, sorry.

http://www.mediafire.com/download/9h5rnc4x6b56xwq/Sprites_Request_11.0.9%282%29.rar

**Comment by MrStonedOne** [comment]
Please make a changelog,

You can add it to the PR's body rather then as a file now, see here: https://tgstation13.org/phpBB/viewtopic.php?f=5&t=5157

**Comment by SweaterKittens** [comment]
@MrStonedOne I'll make sure that a complete changelog is added when this is ready to merge.

**Comment by as334** [comment]
Yeah i know still merge conflicts, i just put up that commit because i have to switch between computers while coding stuff and i had to move the more updated shit over. Progress continues however

**Comment by Cheridan** [comment]
this is a horrible idea



:+1:

**Comment by Incoming5643** [comment]
341s6

**Comment by Lumipharon** [comment]
Wat.

**Comment by Shadowlight213** [comment]
This is almost as broken as goofmed.

**Comment by SweaterKittens** [comment]
@Shadowlight213 Not even remotely, this has been given a lot more thought and input via feedback than goofmed. There was a crippling bug regarding suits that is being worked on right now, but comparing this to /that/ mess which was pumped out in 8 hours is, frankly, insulting.

**Comment by Shadowlight213** [comment]
I said almost. Almost has a wide range.

**Comment by Lumipharon** [comment]
Why are we adding round start infernogriefmen with round start eva suits with rad immunity?

**Comment by SweaterKittens** [comment]
>See This post for details. Please direct non-code related feedback over there as well
Jesus fuck Lumi read the fucking thread, I'm not answering the same issue for the hundredth time

**Comment by Lumipharon** [comment]
But it doesn't answer the question at all.

It's purely benefits (roundstart space suit, bio protection, rad protection, no need for food, can murder people by removing their helmet) except for slowdown, which can be countered for the entire round with 5 minutes in chemistry.
Also unless it's changed at somepoint, spread fire stacks splits the number evenly between you and the victim - the difference is the victim doesn't have a magical auto-extinguisher. Also you can just have a bottle full of patches and infinitely heal your burns anyway.

Basically as a chemist with less then 10 minutes of investment I could just run at sanic speed with infinite fire rape.

**Comment by Shadowlight213** [comment]
@Lumipharon Actually the plasmaman suit isn't space proof.

**Comment by SweaterKittens** [comment]
Which is literally one of the first things addressed in the thread, which I've directed to you to, but you insist on continuing to post here. 

We'll be toning up the damage Plasmamen take from being on fire since taking off your suit to set others on fire is a bit too strong at the moment.

**Comment by Lumipharon** [comment]
>Currently, their "Plasma Suit" is equivalent to an EVA suit, provides 100 bio protection, and is space-worthy

Straight from the OP of the thread, not my fault there's out of date info there.

**Comment by SweaterKittens** [comment]
@Lumipharon If you had actually /read/ the OP, you'd see that I was talking about how they existed prior to my intended changes, and go on to say that they will be losing their EVA-capable suit. Seriously, Lumi. It's pretty obvious if you read on that I've scrapped that particular element, not to mention the fact  that they start with a jumpsuit instead of an exo now. Just in case other people have an issue with not reading the full OP, I've changed the title of that section to prevent further confusion.
